db/background_mgr.go

```diff
@@ -52,47 +76,160 @@ type BackgroundManagerStatus struct {
 // BackgroundManagerProcessI is an interface satisfied by any of the background processes
 // Examples of this: ReSync, Compaction
 type BackgroundManagerProcessI interface {
+	Init(clusterStatus []byte) error
 	Run(options map[string]interface{}, terminator chan struct{}) error
 	GetProcessStatus(status BackgroundManagerStatus) ([]byte, error)
 	ResetStatus()
 }
 
 func (b *BackgroundManager) Start(options map[string]interface{}) error {
+	b.terminator = make(chan struct{}, 1)
+	b.terminatorClosed = false
+
 	err := b.markStart()
 	if err != nil {
 		return err
 	}
 
+	var processClusterStatus []byte
+	if b.clusterAwareOptions != nil {
+		processClusterStatus, _, err = b.clusterAwareOptions.bucket.GetRaw(b.clusterAwareOptions.StatusDocID())
+		if err != nil && !base.IsDocNotFoundError(err) {
+			return err
+		}
+	}
+
 	b.resetStatus()
+
+	err = b.Process.Init(processClusterStatus)
+	if err != nil {
```

Comment:
Init returns error if it fails to unmarshal the processClusterStatus.  In the event of malformed processClusterStatus, how does the user recover?

Reply:
Yeah thats true. Can probably just act as if there was no cluster status and restart rather than resume the process.